tests/transforms/test_transform_can_be_hoisted.py

```diff
@@ -0,0 +1,138 @@
+    assert can_be_hoisted(op2, for_op.body)
+
+
+def test_for_with_loop_invariant_verify():
+    """Test for with loop-variant variables"""
+
+    lower = Constant.from_int_and_width(0, IndexType())
+    upper = Constant.from_int_and_width(42, IndexType())
+    step = Constant.from_int_and_width(3, IndexType())
+    carried = Constant.from_int_and_width(1, IndexType())
+
+    @Builder.implicit_region((IndexType(), IndexType()))
+    def body(_: tuple[BlockArgument, ...]) -> None:
+        a = Constant(IntegerAttr.from_int_and_width(1, 32), i32)
+        b = Constant(IntegerAttr.from_int_and_width(2, 32), i32)
+        # Operations on these constants
+        c = Addi(a, b)
+        d = Muli(a, b)
+        e = DivUI(c, d)
+        Yield(e)
+
+    for_op = For(lower, upper, step, [carried], body)
+
+    for op in for_op.body.walk():
+        for regoin in for_op.regions:
+            # MLIR iterates the for loop's body and hoists operations that
+            # can_be_hoisted. However, the DivUI operation cannot be hoisted immediately
+            # because its operands are still within the loop's body. It can only be
+            # hoisted once the instructions generating those operands have been hoisted.
+            if not (isinstance(op, Yield) or isinstance(op, DivUI)):
+                assert can_be_hoisted(op, regoin)
+
+
+def test_invariant_loop_dialect():
+    """Test for with loop-variant variables"""
+    # Create constants
+    ci0 = Constant.from_int_and_width(0, IndexType())
+    cf7 = Constant(FloatAttr(7.0, f32))
+    cf8 = Constant(FloatAttr(8.0, f32))
+    ci10 = Constant.from_int_and_width(10, IndexType())
+    ci1 = Constant.from_int_and_width(1, IndexType())
+    co0 = Constant.from_int_and_width(0, IndexType())
+    co1 = Constant.from_int_and_width(15, IndexType())
+    coi = Constant.from_int_and_width(1, IndexType())
+
+    @Builder.implicit_region((builtin.IndexType(),))
+    def inner_loop_body(args: tuple[BlockArgument, ...]):
+        bla = Constant.from_int_and_width(5, i32)
+        # Test operation inside the loop
+        hello = test.TestOp([bla], result_types=[i32])
+        test.TestOp([hello], result_types=[])
+        # Floating-point addition inside the loop
+        _v0 = Addf(cf7, cf8)
+
+    @Builder.implicit_region((builtin.IndexType(),))
+    # @Builder.implicit_region((IndexType(),))
+    def outer_loop_body(args: tuple[BlockArgument, ...]):
+        scf.For(ci0, ci10, ci1, [], inner_loop_body)
+
+    for_op = For(co0, co1, coi, [], outer_loop_body)
+    body0 = Block([cf7, cf8, for_op])
+    region0 = Region(body0)
+    # regoin1 = Region([body0, outer_loop_body.block])
+    # # Create the function
+    func_type = FunctionType.from_lists([], [])
+    function = func.FuncOp("invariant_loop_dialect", func_type, region0)
+
+    # Wrap all in a ModuleOp
+    mod = ModuleOp([function])
+    # Assertions to check loop invariant and hoisting
+    for op in mod.body.walk():
+        for regoin in mod.regions:
```

Comment:
```suggestion
        for region in mod.regions:
```
here and in some other places